Add `@customResolver` `requires` to migration guide (#3805)
neo4j/graphql: docs/modules/ROOT/pages/guides/v4-migration/index.adoc

});
----

==== `requires` changes

In version 4.0.0, it is now possible to require non-scalar fields. This means it is also possible to require fields on related type.

> To make this possible, the `requires` argument now accept a graphql selection set instead of a list of strings.

Reviewer: ```suggestion
To make this possible, the `requires` argument now accept a GraphQL selection set instead of a list of strings.
```

> Therefore, the following type definitions:

Reviewer: ```suggestion
Therefore, given the following type definitions in 3.0.0:
```

[source, graphql, indent=0]
----
type User {
    firstName: String!
    lastName: String!
    fullName: String! @customResolver(requires: ["firstName", "lastName"])
}
----

> Would need to be modified to use a selection set as below:

Reviewer: ```suggestion
In 4.0.0, these need to be modified to use a selection set as below:
```
